File: src/test/java/com/benjaminsproule/swagger/gradleplugin/test/java/jaxrs/ExtendedTestResourceWithClassAnnotation.java
package com.benjaminsproule.swagger.gradleplugin.test.java.jaxrs;

import io.swagger.annotations.ApiOperation;

import javax.ws.rs.GET;
import javax.ws.rs.Path;

public class ExtendedTestResourceWithClassAnnotation extends TestResourceWithClassAnnotation {

    @ApiOperation("An overriden operation description")
    @Path("/overriden")
    @GET
    @Override
    public String overriden() {
        return "";
    }

    @Path("/overridenWithoutDescription")
    @GET
    public String overridenWithoutDescription() {
        return "";
    }
}

File: src/test/kotlin/com/benjaminsproule/swagger/gradleplugin/test/kotlin/jaxrs/ExtendedTestResourceWithClassAnnotation.kt
package com.benjaminsproule.swagger.gradleplugin.test.kotlin.jaxrs

import io.swagger.annotations.ApiOperation

import javax.ws.rs.GET
import javax.ws.rs.Path

class ExtendedTestResourceWithClassAnnotation : TestResourceWithClassAnnotation() {

    @ApiOperation("An overriden operation description")
    @Path("/overriden")
    @GET
    override fun overriden(): String {
        return ""
    }

    @Path("/overridenWithoutDescription")
    @GET
    override fun overridenWithoutDescription(): String {
        return ""
    }
}
